presto-main/src/main/java/com/facebook/presto/execution/scheduler/NodeScheduler.java

```diff
@@ -306,8 +311,8 @@ public static SplitPlacementResult selectDistributionNodes(
             boolean isCacheable = bucketNodeMap.isSplitCacheable(split);
 
             // if node is full, don't schedule now, which will push back on the scheduling of splits
-            if (assignmentStats.getTotalSplitCount(node) < maxSplitsPerNode ||
-                    assignmentStats.getQueuedSplitCountForStage(node) < maxPendingSplitsPerTask) {
+            if (assignmentStats.getUnacknowledgedSplitCountForStage(node) < maxUnacknowledgedSplitsPerTask &&
```

Comment:
This is getting hard to read. Maybe extract it to a standalone function `isNodeFull` that returns `boolean`?

Reply:
Good idea, done.